### DISCUSSION TITLE
🏈 Remastered GM

### DISCUSSION THREAD

**Discussion by swiftyfinch** [swiftyfinch/Rugby#229] [Releases]
<p align="center">
  <img src="https://user-images.githubusercontent.com/64660122/230763146-d467dafb-252c-46ce-93d6-f2309e5aabc8.gif" width=600>
</p>

Use this command to update Rugby from `2.0.0b2+`:
```
rugby update --beta
```
Please, check out the 📖 [readme](https://github.com/swiftyfinch/Rugby/tree/2.0.0b9#readme), you can find there all information.
The previous release is [here](https://github.com/swiftyfinch/Rugby/releases/tag/2.0.0b8).

## Changelog

- 🐛 Added the full Swift version to hash algorithm https://github.com/swiftyfinch/Rugby/discussions/226#discussioncomment-5465831;
- 🐛 Fixed issue with `$(inherited)` keys in hash algorithm;
- 🐛 Handled a missing Xcode CLT;
- ⛳️ Added flag `--delete-sources` as it was in the Rugby1.x;
- 🚑 Added `doctor` command as it was in the Rugby1.x;
- 📊 Stopped sending analytics;
- 📖 Added more docs;
- 🏛️ Removed the `auto` architecture and used `arm64` by default.

<br>

<hr /><em>This discussion was created from the release <a href='https://github.com/swiftyfinch/Rugby/releases/tag/2.0.0b9'>🏈 Remastered GM</a>.</em>

**Comment by zzzworm** [comment]
Please add --no-graph support for 2.0!
Sometime we change a low level pod's code will not affect the parent pods, but all other pods depend on it will rebuild, this very slow.
Also will the low level pod's binary cache used by parent pods when call `rugby build`?

**Comment by swiftyfinch** [comment]
Hello @zzzworm!

Yeah, the [build](https://github.com/swiftyfinch/Rugby/blob/2.0.0b9/docs/commands-help/build.md) command doesn't rebuild targets if you build them earlier.
And it reuses them from the `~/.rugby/bin` folder.

<img width="347" alt="Screenshot 2023-04-11 at 15 56 30" src="https://user-images.githubusercontent.com/64660122/231169923-72739fe5-0525-4694-b6b4-5a7da185fee8.png">

You can check your build logs to be sure. Find them here `~/.rugby/logs`.

<br>

> Please add --no-graph support for 2.0!
Sometime we change a low level pod's code will not affect the parent pods, but all other pods depend on it will rebuild, this very slow.

Oh, I'm surprised that somebody is using this flag. It was buggy, and I got negative feedback about it.
The problem is - we can't be sure that changes don't affect parent targets.
If we can be sure about it, I will add this flag. But from my perspective, I don't see any stable solution to this problem.

BTW, thanks a lot for your feedback!
